src/main/java/org/spongepowered/api/entity/Entity.java

```diff
@@ -415,4 +418,20 @@ default boolean damage(double damage, DamageSource damageSource) {
         return getWorld().getEntities(predicate::test);
     }
 
+    /**
+     * Gets the current display name for this {@link Entity}.
+     *
+     * @return The display name
+     */
+    Text getDisplayName();
```

Comment:
@gabizou The "custom name tag" is used to construct the display name.
This method:
- will return the value of `Keys.DISPLAY_NAME` if present
- - will return the contextual value if using the `ContextViewer` method
- will return `net.minecraft.entity.Entity#getDisplayName` otherwise


`Entity`:
```java
    public String getName()
    {
        if (this.hasCustomName())
        {
            return this.getCustomNameTag();
        }
        else
        {
            String s = EntityList.getEntityString(this);

            if (s == null)
            {
                s = "generic";
            }

            return StatCollector.translateToLocal("entity." + s + ".name");
        }
    }

    public IChatComponent getDisplayName()
    {
        ChatComponentText chatcomponenttext = new ChatComponentText(this.getName());
        chatcomponenttext.getChatStyle().setChatHoverEvent(this.getHoverEvent());
        chatcomponenttext.getChatStyle().setInsertion(this.getUniqueID().toString());
        return chatcomponenttext;
    }
```